metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/usage.py

```diff
@@ -0,0 +1,784 @@
+from datahub.ingestion.source.bigquery_v2.bigquery_audit import (
+    BQ_AUDIT_V2,
+    AuditEvent,
+    AuditLogEntry,
+    BigQueryAuditMetadata,
+    BigQueryTableRef,
+    QueryEvent,
+    ReadEvent,
+)
+from datahub.ingestion.source.bigquery_v2.bigquery_config import BigQueryV2Config
+from datahub.ingestion.source.bigquery_v2.bigquery_report import BigQueryV2Report
+from datahub.ingestion.source.bigquery_v2.common import (
+    BQ_DATE_SHARD_FORMAT,
+    BQ_DATETIME_FORMAT,
+    _make_gcp_logging_client,
+)
+from datahub.ingestion.source.usage.usage_common import GenericAggregatedDataset
+from datahub.metadata.schema_classes import OperationClass, OperationTypeClass
+from datahub.utilities.delayed_iter import delayed_iter
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+AggregatedDataset = GenericAggregatedDataset[BigQueryTableRef]
+
+OPERATION_STATEMENT_TYPES = {
+    "INSERT": OperationTypeClass.INSERT,
+    "UPDATE": OperationTypeClass.UPDATE,
+    "DELETE": OperationTypeClass.DELETE,
+    "MERGE": OperationTypeClass.UPDATE,
+    "CREATE": OperationTypeClass.CREATE,
+    "CREATE_TABLE_AS_SELECT": OperationTypeClass.CREATE,
+    "CREATE_SCHEMA": OperationTypeClass.CREATE,
+    "DROP_TABLE": OperationTypeClass.DROP,
+}
+
+READ_STATEMENT_TYPES: List[str] = ["SELECT"]
+
+
+def bigquery_audit_metadata_query_template(
+    dataset: str,
+    use_date_sharded_tables: bool,
+    table_allow_filter: str = None,
+    limit: Optional[int] = None,
+) -> str:
+    """
+    Receives a dataset (with project specified) and returns a query template that is used to query exported
+    v2 AuditLogs containing protoPayloads of type BigQueryAuditMetadata.
+    :param dataset: the dataset to query against in the form of $PROJECT.$DATASET
+    :param use_date_sharded_tables: whether to read from date sharded audit log tables or time partitioned audit log
+           tables
+    :param table_allow_filter: regex used to filter on log events that contain the wanted datasets
+    :return: a query template, when supplied start_time and end_time, can be used to query audit logs from BigQuery
+    """
+    allow_filter = f"""
+      AND EXISTS (SELECT *
+              from UNNEST(JSON_EXTRACT_ARRAY(protopayload_auditlog.metadataJson,
+                                             "$.jobChange.job.jobStats.queryStats.referencedTables")) AS x
+              where REGEXP_CONTAINS(x, r'(projects/.*/datasets/.*/tables/{table_allow_filter if table_allow_filter else ".*"})'))
+    """
+
+    query: str
+    if use_date_sharded_tables:
+        query = (
+            f"""
+        SELECT
+            timestamp,
+            logName,
+            insertId,
+            protopayload_auditlog AS protoPayload,
+            protopayload_auditlog.metadataJson AS metadata
+        FROM
+            `{dataset}.cloudaudit_googleapis_com_data_access_*`
+        """
+            + """
+        WHERE
+            _TABLE_SUFFIX BETWEEN "{start_date}" AND "{end_date}"
```

Comment:
Looks like you forgot to change `start_date` and `end_date` format place holders with `start_time` and `end_time` in the first query.